internal/processing/istio/reconciliation.go

```diff
@@ -0,0 +1,48 @@
+package istio
+
+import (
+	"context"
+
+	gatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"github.com/kyma-incubator/api-gateway/internal/processing"
+	"github.com/kyma-incubator/api-gateway/internal/validation"
+	networkingv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Reconciliation struct {
+	processors []processing.ReconciliationProcessor
+	config     processing.ReconciliationConfig
+}
+
+func NewIstioReconciliation(config processing.ReconciliationConfig) Reconciliation {
+	vsProcessor := NewVirtualServiceProcessor(config)
+	apProcessor := NewAuthorizationPolicyProcessor(config)
+	raProcessor := NewRequestAuthenticationProcessor(config)
+
+	return Reconciliation{
+		processors: []processing.ReconciliationProcessor{vsProcessor, raProcessor, apProcessor},
```

Comment:
The access rule processor will be added in this task:
- https://github.com/kyma-project/api-gateway/issues/96